File: frontend/lib/metric-meta.ts
import type { MetricKind } from './api';

export const METRIC_META: Record<
  MetricKind,
  { label: string; unit: string; short: string; emoji: string; color: string }
> = {
  pulse: { label: 'Пульс', short: 'Пульс', unit: 'уд/мин', emoji: '💓', color: '#e11d48' },
  bp_sys: { label: 'Давление (верхнее)', short: 'АД верх.', unit: 'мм рт.ст.', emoji: '🩺', color: '#0d9488' },
  bp_dia: { label: 'Давление (нижнее)', short: 'АД ниж.', unit: 'мм рт.ст.', emoji: '🩺', color: '#0f766e' },
  glucose: { label: 'Глюкоза', short: 'Сахар', unit: 'ммоль/л', emoji: '🩸', color: '#ea580c' },
  temperature: { label: 'Температура', short: 'Темп.', unit: '°C', emoji: '🌡️', color: '#f59e0b' },
  weight: { label: 'Вес', short: 'Вес', unit: 'кг', emoji: '⚖️', color: '#64748b' },
  spo2: { label: 'Сатурация', short: 'SpO₂', unit: '%', emoji: '🫁', color: '#14b8a6' },
};

export function formatValue(kind: MetricKind, v: number): string {
  if (kind === 'glucose') return v.toFixed(1);
  if (kind === 'temperature') return v.toFixed(1);
  if (kind === 'weight') return v.toFixed(1);
  return Math.round(v).toString();
}

File: mobile/lib/widgets/metric_meta.dart
// Per-metric display metadata: emoji, unit, value formatter, normal range
// hint. Mirrors `frontend/lib/metric-meta.ts` so the UI stays consistent
// with what the backend reasons about.
import "package:flutter/material.dart";

class MetricMeta {
  const MetricMeta({
    required this.kind,
    required this.emoji,
    required this.unit,
    required this.color,
    required this.fmt,
    required this.suggestedRange,
  });
  final String kind;
  final String emoji;
  final String unit;
  final Color color;
  final String Function(double) fmt;
  final (double, double) suggestedRange;
}

const _teal = Color(0xFF0F766E);
const _amber = Color(0xFFD97706);
const _rose = Color(0xFFE11D48);
const _indigo = Color(0xFF4F46E5);

final Map<String, MetricMeta> metricMeta = {
  "pulse": MetricMeta(
    kind: "pulse",
    emoji: "❤️",
    unit: "уд/мин",
    color: _rose,
    fmt: (v) => v.round().toString(),
    suggestedRange: (40, 140),
  ),
  "bp_sys": MetricMeta(
    kind: "bp_sys",
    emoji: "🩸",
    unit: "мм рт.ст.",
    color: _rose,
    fmt: (v) => v.round().toString(),
    suggestedRange: (80, 200),
  ),
  "bp_dia": MetricMeta(
    kind: "bp_dia",
    emoji: "🩸",
    unit: "мм рт.ст.",
    color: _rose,
    fmt: (v) => v.round().toString(),
    suggestedRange: (50, 120),
  ),
  "glucose": MetricMeta(
    kind: "glucose",
    emoji: "🩸",
    unit: "ммоль/л",
    color: _amber,
    fmt: (v) => v.toStringAsFixed(1),
    suggestedRange: (3, 18),
  ),
  "temperature": MetricMeta(
    kind: "temperature",
    emoji: "🌡",
    unit: "°C",
    color: _indigo,
    fmt: (v) => v.toStringAsFixed(1),
    suggestedRange: (35, 41),
  ),
  "weight": MetricMeta(
    kind: "weight",
    emoji: "⚖️",
    unit: "кг",
    color: _teal,
    fmt: (v) => v.toStringAsFixed(1),
    suggestedRange: (30, 200),
  ),
  "spo2": MetricMeta(
    kind: "spo2",
    emoji: "💨",
    unit: "%",
    color: _teal,
    fmt: (v) => v.round().toString(),
    suggestedRange: (80, 100),
  ),
};
